### ISSUE TITLE
stack does not deploy correctly. pgpool does not connect to backends, and db does not initialize

### ISSUE THREAD

**Issue by ksonda** [bitnami/bitnami-docker-pgpool#38]
**Description**

I have attempted to deploy the stack with the stock docker-compose.yml in this repository, both with an empty database and with a database defined in an .sql script with a volume mapped to the docker-entrypoint-initdb,d in the docker-compose.yml. pgpool is not able to connect to the backend dbs, 

**Steps to reproduce the issue:**

1. git clone https://github.com/ksonda/pidpool
2. cd pidpool
3. docker-compose up

**Describe the results you received:**
```
Creating network "pidpool_default" with the default driver
Creating volume "pidpool_pg_1_data" with local driver
Creating volume "pidpool_pg_0_data" with local driver
Pulling pg-0 (docker.io/bitnami/postgresql-repmgr:11-debian-10)...
11-debian-10: Pulling from bitnami/postgresql-repmgr
522a7372a4ae: Pull complete
6e99d8be05c9: Pull complete
1a930db656e9: Pull complete
6ebf3a542aea: Pull complete
2841add8fe49: Pull complete
acb71b7b612a: Pull complete
19cfc60a2200: Pull complete
ed5628af8e6d: Pull complete
e6392ca95a08: Pull complete
4cfde8944dd6: Pull complete
89d8d1260ba6: Pull complete
12d5aa3371f2: Pull complete
6fe050522d1b: Pull complete
9b46d8ba1d06: Pull complete
Digest: sha256:25dfe20922d55dd715e3c19a4a0d217912fff57c007942f926b9e7def4d6e634
Status: Downloaded newer image for bitnami/postgresql-repmgr:11-debian-10
Pulling pgpool (docker.io/bitnami/pgpool:4-debian-10)...
4-debian-10: Pulling from bitnami/pgpool
522a7372a4ae: Already exists
79d7ea45ce59: Pull complete
29ea64266b2a: Pull complete
cd79ca72fd06: Pull complete
7afed88b2fab: Pull complete
9499230b85aa: Pull complete
187329433ef3: Pull complete
7bd99a4f100f: Pull complete
2d94afed0884: Pull complete
25b501564c80: Pull complete
23187f6e9fc2: Pull complete
Digest: sha256:038aab4040bf7c4c546afc4799d0ac4b259fcf33a61d26abd94cfbfe087885d9
Status: Downloaded newer image for bitnami/pgpool:4-debian-10
Creating pidpool_pg-0_1 ... 
Creating pidpool_pg-1_1 ... 
Creating pidpool_pgpool_1 ... 
Creating pidpool_pgpool_1
Creating pidpool_pg-0_1
Creating pidpool_pg-1_1 ... done
Attaching to pidpool_pg-0_1, pidpool_pgpool_1, pidpool_pg-1_1
pg-0_1    | postgresql-repmgr 16:17:01.32 
pgpool_1  | pgpool 16:17:01.39 
pg-0_1    | postgresql-repmgr 16:17:01.32 Welcome to the Bitnami postgresql-repmgr container
pg-0_1    | postgresql-repmgr 16:17:01.32 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-postgresql-repmgr
pgpool_1  | pgpool 16:17:01.39 Welcome to the Bitnami pgpool container
pg-0_1    | postgresql-repmgr 16:17:01.32 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-postgresql-repmgr/issues
pg-1_1    | postgresql-repmgr 16:17:02.31 
pg-0_1    | postgresql-repmgr 16:17:01.33 
pgpool_1  | pgpool 16:17:01.39 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-pgpool
pgpool_1  | pgpool 16:17:01.39 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-pgpool/issues
pg-0_1    | postgresql-repmgr 16:17:01.38 INFO  ==> ** Starting PostgreSQL with Replication Manager setup **
pg-0_1    | postgresql-repmgr 16:17:01.42 INFO  ==> Validating settings in REPMGR_* env vars...
pg-0_1    | postgresql-repmgr 16:17:01.42 INFO  ==> Validating settings in POSTGRESQL_* env vars..
pg-0_1    | postgresql-repmgr 16:17:01.43 INFO  ==> Querying all partner nodes for common upstream node...
pgpool_1  | pgpool 16:17:01.40 
pgpool_1  | pgpool 16:17:01.40 INFO  ==> ** Starting Pgpool-II setup **
pgpool_1  | pgpool 16:17:01.43 INFO  ==> Validating settings in PGPOOL_* env vars...
pgpool_1  | pgpool 16:17:01.46 INFO  ==> Initializing Pgpool-II...
pgpool_1  | pgpool 16:17:01.47 INFO  ==> Generating pg_hba.conf file...
pgpool_1  | pgpool 16:17:01.48 INFO  ==> Generating pgpool.conf file...
pgpool_1  | pgpool 16:17:01.62 INFO  ==> Generating password file for local authentication...
pgpool_1  | pgpool 16:17:01.63 INFO  ==> Generating password file for pgpool admin user...
pgpool_1  | 
pgpool_1  | pgpool 16:17:01.64 INFO  ==> ** Pgpool-II setup finished! **
pgpool_1  | pgpool 16:17:01.66 INFO  ==> ** Starting Pgpool-II **
pgpool_1  | 2020-09-25 16:17:01: pid 1: LOG:  Backend status file /opt/bitnami/pgpool/logs/pgpool_status does not exist
pgpool_1  | 2020-09-25 16:17:01: pid 1: LOG:  memory cache initialized
pgpool_1  | 2020-09-25 16:17:01: pid 1: DETAIL:  memcache blocks :64
pgpool_1  | 2020-09-25 16:17:01: pid 1: LOG:  pool_discard_oid_maps: discarded memqcache oid maps
pgpool_1  | 2020-09-25 16:17:01: pid 1: LOG:  Setting up socket for 0.0.0.0:5432
pgpool_1  | 2020-09-25 16:17:01: pid 1: LOG:  Setting up socket for :::5432
pgpool_1  | 2020-09-25 16:17:01: pid 1: LOG:  find_primary_node_repeatedly: waiting for finding a primary node
pgpool_1  | 2020-09-25 16:17:01: pid 1: LOG:  failed to connect to PostgreSQL server on "pg-0:5432", getsockopt() detected error "Connection refused"
pgpool_1  | 2020-09-25 16:17:01: pid 1: ERROR:  failed to make persistent db connection
pgpool_1  | 2020-09-25 16:17:01: pid 1: DETAIL:  connection to host:"pg-0:5432" failed
pgpool_1  | 2020-09-25 16:17:01: pid 1: LOG:  find_primary_node: make_persistent_db_connection_noerror failed on node 0
pg-1_1    | postgresql-repmgr 16:17:02.31 Welcome to the Bitnami postgresql-repmgr container
pg-1_1    | postgresql-repmgr 16:17:02.31 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-postgresql-repmgr
pg-1_1    | postgresql-repmgr 16:17:02.32 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-postgresql-repmgr/issues
pg-1_1    | postgresql-repmgr 16:17:02.32 
pg-1_1    | postgresql-repmgr 16:17:02.34 INFO  ==> ** Starting PostgreSQL with Replication Manager setup **
pg-1_1    | postgresql-repmgr 16:17:02.37 INFO  ==> Validating settings in REPMGR_* env vars...
pg-1_1    | postgresql-repmgr 16:17:02.37 INFO  ==> Validating settings in POSTGRESQL_* env vars..
pg-1_1    | postgresql-repmgr 16:17:02.38 INFO  ==> Querying all partner nodes for common upstream node...
pgpool_1  | 2020-09-25 16:17:02: pid 1: LOG:  failed to connect to PostgreSQL server on "pg-1:5432", getsockopt() detected error "Connection refused"
pgpool_1  | 2020-09-25 16:17:02: pid 1: ERROR:  failed to make persistent db connection
pgpool_1  | 2020-09-25 16:17:02: pid 1: DETAIL:  connection to host:"pg-1:5432" failed
pgpool_1  | 2020-09-25 16:17:02: pid 1: LOG:  find_primary_node: make_persistent_db_connection_noerror failed on node 1
pg-0_1    | postgresql-repmgr 16:17:02.41 INFO  ==> There are no nodes with primary role. Assuming the primary role...
pg-0_1    | postgresql-repmgr 16:17:02.42 INFO  ==> Preparing PostgreSQL configuration...
pg-0_1    | postgresql-repmgr 16:17:02.42 INFO  ==> postgresql.conf file not detected. Generating it...
pg-1_1    | postgresql-repmgr 16:17:02.44 INFO  ==> Preparing PostgreSQL configuration...
pg-1_1    | postgresql-repmgr 16:17:02.44 INFO  ==> postgresql.conf file not detected. Generating it...
pg-0_1    | postgresql-repmgr 16:17:02.55 INFO  ==> Preparing repmgr configuration...
pg-0_1    | postgresql-repmgr 16:17:02.56 INFO  ==> Initializing Repmgr...
pg-1_1    | postgresql-repmgr 16:17:02.57 INFO  ==> Preparing repmgr configuration...
pg-0_1    | postgresql-repmgr 16:17:02.57 INFO  ==> Initializing PostgreSQL database...
pg-1_1    | postgresql-repmgr 16:17:02.58 INFO  ==> Initializing Repmgr...
pg-0_1    | postgresql-repmgr 16:17:02.59 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/postgresql.conf detected
pg-1_1    | postgresql-repmgr 16:17:02.59 INFO  ==> Waiting for primary node...
pg-0_1    | postgresql-repmgr 16:17:02.59 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/pg_hba.conf detected
pgpool_1  | 2020-09-25 16:17:03: pid 1: LOG:  failed to connect to PostgreSQL server on "pg-0:5432", getsockopt() detected error "Connection refused"
pgpool_1  | 2020-09-25 16:17:03: pid 1: ERROR:  failed to make persistent db connection
pgpool_1  | 2020-09-25 16:17:03: pid 1: DETAIL:  connection to host:"pg-0:5432" failed
pgpool_1  | 2020-09-25 16:17:03: pid 1: LOG:  find_primary_node: make_persistent_db_connection_noerror failed on node 0
pgpool_1  | 2020-09-25 16:17:03: pid 1: LOG:  failed to connect to PostgreSQL server on "pg-1:5432", getsockopt() detected error "Connection refused"
pgpool_1  | 2020-09-25 16:17:03: pid 1: ERROR:  failed to make persistent db connection
pgpool_1  | 2020-09-25 16:17:03: pid 1: DETAIL:  connection to host:"pg-1:5432" failed
pgpool_1  | 2020-09-25 16:17:03: pid 1: LOG:  find_primary_node: make_persistent_db_connection_noerror failed on node 1
pg-0_1    | postgresql-repmgr 16:17:04.03 INFO  ==> Starting PostgreSQL in background...
pgpool_1  | 2020-09-25 16:17:04: pid 1: LOG:  failed to connect to PostgreSQL server on "pg-0:5432", getsockopt() detected error "Connection refused"
pgpool_1  | 2020-09-25 16:17:04: pid 1: ERROR:  failed to make persistent db connection
pgpool_1  | 2020-09-25 16:17:04: pid 1: DETAIL:  connection to host:"pg-0:5432" failed
pgpool_1  | 2020-09-25 16:17:04: pid 1: LOG:  find_primary_node: make_persistent_db_connection_noerror failed on node 0
pgpool_1  | 2020-09-25 16:17:04: pid 1: LOG:  failed to connect to PostgreSQL server on "pg-1:5432", getsockopt() detected error "Connection refused"
pgpool_1  | 2020-09-25 16:17:04: pid 1: ERROR:  failed to make persistent db connection
pgpool_1  | 2020-09-25 16:17:04: pid 1: DETAIL:  connection to host:"pg-1:5432" failed
pgpool_1  | 2020-09-25 16:17:04: pid 1: LOG:  find_primary_node: make_persistent_db_connection_noerror failed on node 1
pg-0_1    | postgresql-repmgr 16:17:04.44 INFO  ==> Changing password of postgres
pg-0_1    | postgresql-repmgr 16:17:04.46 INFO  ==> Creating user customuser
pg-0_1    | postgresql-repmgr 16:17:04.48 INFO  ==> Granting access to "customuser" to the database "customdatabase"
pg-0_1    | postgresql-repmgr 16:17:04.49 INFO  ==> Creating replication user repmgr
pg-0_1    | postgresql-repmgr 16:17:04.53 INFO  ==> Configuring replication parameters
pg-0_1    | postgresql-repmgr 16:17:04.58 INFO  ==> Configuring fsync
pg-0_1    | postgresql-repmgr 16:17:04.59 INFO  ==> Starting PostgreSQL in background...
pg-0_1    | postgresql-repmgr 16:17:04.59 INFO  ==> Creating repmgr user: repmgr
pg-0_1    | postgresql-repmgr 16:17:04.64 INFO  ==> Creating repmgr database: repmgr
pg-0_1    | postgresql-repmgr 16:17:04.91 INFO  ==> Stopping PostgreSQL...
pgpool_1  | 2020-09-25 16:17:05: pid 1: LOG:  failed to connect to PostgreSQL server on "pg-0:5432", getsockopt() detected error "Connection refused"
pgpool_1  | 2020-09-25 16:17:05: pid 1: ERROR:  failed to make persistent db connection
pgpool_1  | 2020-09-25 16:17:05: pid 1: DETAIL:  connection to host:"pg-0:5432" failed
pgpool_1  | 2020-09-25 16:17:05: pid 1: LOG:  find_primary_node: make_persistent_db_connection_noerror failed on node 0
pgpool_1  | 2020-09-25 16:17:05: pid 1: LOG:  failed to connect to PostgreSQL server on "pg-1:5432", getsockopt() detected error "Connection refused"
pgpool_1  | 2020-09-25 16:17:05: pid 1: ERROR:  failed to make persistent db connection
pgpool_1  | 2020-09-25 16:17:05: pid 1: DETAIL:  connection to host:"pg-1:5432" failed
pgpool_1  | 2020-09-25 16:17:05: pid 1: LOG:  find_primary_node: make_persistent_db_connection_noerror failed on node 1
pg-0_1    | postgresql-repmgr 16:17:05.92 INFO  ==> Starting PostgreSQL in background...
pg-0_1    | postgresql-repmgr 16:17:06.04 INFO  ==> Registering Primary...
pg-0_1    | postgresql-repmgr 16:17:06.17 INFO  ==> Loading custom scripts...
pg-0_1    | postgresql-repmgr 16:17:06.18 INFO  ==> Stopping PostgreSQL...
pgpool_1  | 2020-09-25 16:17:06: pid 1: LOG:  failed to connect to PostgreSQL server on "pg-0:5432", getsockopt() detected error "Connection refused"
pgpool_1  | 2020-09-25 16:17:06: pid 1: ERROR:  failed to make persistent db connection
pgpool_1  | 2020-09-25 16:17:06: pid 1: DETAIL:  connection to host:"pg-0:5432" failed
pgpool_1  | 2020-09-25 16:17:06: pid 1: LOG:  find_primary_node: make_persistent_db_connection_noerror failed on node 0
pgpool_1  | 2020-09-25 16:17:06: pid 1: LOG:  failed to connect to PostgreSQL server on "pg-1:5432", getsockopt() detected error "Connection refused"
pgpool_1  | 2020-09-25 16:17:06: pid 1: ERROR:  failed to make persistent db connection
pgpool_1  | 2020-09-25 16:17:06: pid 1: DETAIL:  connection to host:"pg-1:5432" failed
pgpool_1  | 2020-09-25 16:17:06: pid 1: LOG:  find_primary_node: make_persistent_db_connection_noerror failed on node 1
pg-0_1    | 
pg-0_1    | postgresql-repmgr 16:17:07.19 INFO  ==> ** PostgreSQL with Replication Manager setup finished! **
pg-0_1    | postgresql-repmgr 16:17:07.22 INFO  ==> Starting PostgreSQL in background...
pg-0_1    | waiting for server to start....2020-09-25 16:17:07.258 GMT [252] LOG:  listening on IPv4 address "0.0.0.0", port 5432
pg-0_1    | 2020-09-25 16:17:07.258 GMT [252] LOG:  listening on IPv6 address "::", port 5432
pg-0_1    | 2020-09-25 16:17:07.265 GMT [252] LOG:  listening on Unix socket "/tmp/.s.PGSQL.5432"
pg-0_1    | 2020-09-25 16:17:07.277 GMT [252] LOG:  redirecting log output to logging collector process
pg-0_1    | 2020-09-25 16:17:07.277 GMT [252] HINT:  Future log output will appear in directory "/opt/bitnami/postgresql/logs".
pg-0_1    | 2020-09-25 16:17:07.283 GMT [254] LOG:  database system was shut down at 2020-09-25 16:17:06 GMT
pg-0_1    | 2020-09-25 16:17:07.290 GMT [252] LOG:  database system is ready to accept connections
pg-0_1    |  done
pg-0_1    | server started
pg-0_1    | postgresql-repmgr 16:17:07.35 INFO  ==> ** Starting repmgrd **
pg-0_1    | [2020-09-25 16:17:07] [NOTICE] repmgrd (repmgrd 5.1.0) starting up
pg-0_1    | INFO:  set_repmgrd_pid(): provided pidfile is /opt/bitnami/repmgr/tmp/repmgr.pid
pg-0_1    | [2020-09-25 16:17:07] [NOTICE] starting monitoring of node "pg-0" (ID: 1000)
pg-0_1    | [2020-09-25 16:17:07] [NOTICE] monitoring cluster primary "pg-0" (ID: 1000)
pgpool_1  | 2020-09-25 16:17:07: pid 1: LOG:  failed to connect to PostgreSQL server on "pg-1:5432", getsockopt() detected error "Connection refused"
pgpool_1  | 2020-09-25 16:17:07: pid 1: ERROR:  failed to make persistent db connection
pgpool_1  | 2020-09-25 16:17:07: pid 1: DETAIL:  connection to host:"pg-1:5432" failed
pgpool_1  | 2020-09-25 16:17:07: pid 1: LOG:  find_primary_node: make_persistent_db_connection_noerror failed on node 1
pgpool_1  | 2020-09-25 16:17:07: pid 1: LOG:  find_primary_node: primary node is 0
pgpool_1  | 2020-09-25 16:17:07: pid 145: LOG:  failed to connect to PostgreSQL server on "pg-1:5432", getsockopt() detected error "Connection refused"
pgpool_1  | 2020-09-25 16:17:07: pid 142: LOG:  PCP process: 142 started
pgpool_1  | 2020-09-25 16:17:07: pid 145: ERROR:  failed to make persistent db connection
pgpool_1  | 2020-09-25 16:17:07: pid 145: DETAIL:  connection to host:"pg-1:5432" failed
pgpool_1  | 2020-09-25 16:17:07: pid 145: LOG:  health check retrying on DB node: 1 (round:1)
pgpool_1  | 2020-09-25 16:17:07: pid 1: LOG:  pgpool-II successfully started. version 4.1.4 (karasukiboshi)
pgpool_1  | 2020-09-25 16:17:07: pid 1: LOG:  node status[0]: 1
pgpool_1  | 2020-09-25 16:17:07: pid 1: LOG:  node status[1]: 0
pgpool_1  | 2020-09-25 16:17:07: pid 143: LOG:  failed to connect to PostgreSQL server on "pg-1:5432", getsockopt() detected error "Connection refused"
pgpool_1  | 2020-09-25 16:17:07: pid 143: ERROR:  failed to make persistent db connection
pgpool_1  | 2020-09-25 16:17:07: pid 143: DETAIL:  connection to host:"pg-1:5432" failed
pgpool_1  | 2020-09-25 16:17:11: pid 140: LOG:  failed to connect to PostgreSQL server on "pg-1:5432", getsockopt() detected error "Connection refused"
pgpool_1  | 2020-09-25 16:17:11: pid 140: LOG:  failed to create a backend 1 connection
pgpool_1  | 2020-09-25 16:17:11: pid 140: DETAIL:  skip this backend because because failover_on_backend_error is off and we are in streaming replication mode and node is standby node
pgpool_1  | 2020-09-25 16:17:11: pid 140: LOG:  pool_reuse_block: blockid: 0
pgpool_1  | 2020-09-25 16:17:11: pid 140: CONTEXT:  while searching system catalog, When relcache is missed
pgpool_1  | 2020-09-25 16:17:11: pid 140: LOG:  failover or failback event detected
pgpool_1  | 2020-09-25 16:17:11: pid 140: DETAIL:  restarting myself
pgpool_1  | 2020-09-25 16:17:11: pid 1: LOG:  child process with pid: 140 exits with status 256
pgpool_1  | 2020-09-25 16:17:11: pid 1: LOG:  fork a new child process with pid: 162
pgpool_1  | 2020-09-25 16:17:11: pid 137: LOG:  failed to connect to PostgreSQL server on "pg-1:5432", getsockopt() detected error "Connection refused"
pgpool_1  | 2020-09-25 16:17:11: pid 137: LOG:  failed to create a backend 1 connection
pgpool_1  | 2020-09-25 16:17:11: pid 137: DETAIL:  skip this backend because because failover_on_backend_error is off and we are in streaming replication mode and node is standby node
pgpool_1  | 2020-09-25 16:17:11: pid 137: LOG:  failover or failback event detected
pgpool_1  | 2020-09-25 16:17:11: pid 137: DETAIL:  restarting myself
pgpool_1  | 2020-09-25 16:17:11: pid 1: LOG:  child process with pid: 137 exits with status 256
pgpool_1  | 2020-09-25 16:17:11: pid 1: LOG:  fork a new child process with pid: 164
pgpool_1  | 2020-09-25 16:17:12: pid 145: LOG:  failed to connect to PostgreSQL server on "pg-1:5432", getsockopt() detected error "Connection refused"
pgpool_1  | 2020-09-25 16:17:12: pid 145: ERROR:  failed to make persistent db connection
pgpool_1  | 2020-09-25 16:17:12: pid 145: DETAIL:  connection to host:"pg-1:5432" failed
pgpool_1  | 2020-09-25 16:17:12: pid 145: LOG:  health check retrying on DB node: 1 (round:2)
pg-1_1    | postgresql-repmgr 16:17:12.63 INFO  ==> Cloning data from primary node...
pg-1_1    | postgresql-repmgr 16:17:13.40 INFO  ==> Initializing PostgreSQL database...
pg-1_1    | postgresql-repmgr 16:17:13.40 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/postgresql.conf detected
pg-1_1    | postgresql-repmgr 16:17:13.40 INFO  ==> Custom configuration /opt/bitnami/postgresql/conf/pg_hba.conf detected
pg-1_1    | postgresql-repmgr 16:17:13.43 INFO  ==> Deploying PostgreSQL with persisted data...
pg-1_1    | postgresql-repmgr 16:17:13.44 INFO  ==> Configuring replication parameters
pg-1_1    | postgresql-repmgr 16:17:13.49 INFO  ==> Configuring fsync
pg-1_1    | postgresql-repmgr 16:17:13.51 INFO  ==> Setting up streaming replication slave...
pg-1_1    | postgresql-repmgr 16:17:13.54 INFO  ==> Starting PostgreSQL in background...
pg-0_1    | 2020-09-25 16:17:13.727 GMT [292] LOG:  standby "walreceiver" is now a synchronous standby with priority 1
pg-1_1    | postgresql-repmgr 16:17:13.76 INFO  ==> Unregistering standby node...
pg-1_1    | postgresql-repmgr 16:17:13.78 INFO  ==> Registering Standby node...
pg-1_1    | postgresql-repmgr 16:17:13.84 INFO  ==> Stopping PostgreSQL...
pg-1_1    | postgresql-repmgr 16:17:14.85 INFO  ==> ** PostgreSQL with Replication Manager setup finished! **
pg-1_1    | 
pg-1_1    | postgresql-repmgr 16:17:14.88 INFO  ==> Starting PostgreSQL in background...
pg-1_1    | waiting for server to start....2020-09-25 16:17:14.910 GMT [193] LOG:  listening on IPv4 address "0.0.0.0", port 5432
pg-1_1    | 2020-09-25 16:17:14.910 GMT [193] LOG:  listening on IPv6 address "::", port 5432
pg-1_1    | 2020-09-25 16:17:14.916 GMT [193] LOG:  listening on Unix socket "/tmp/.s.PGSQL.5432"
pg-1_1    | 2020-09-25 16:17:14.928 GMT [193] LOG:  redirecting log output to logging collector process
pg-1_1    | 2020-09-25 16:17:14.928 GMT [193] HINT:  Future log output will appear in directory "/opt/bitnami/postgresql/logs".
pg-1_1    | 2020-09-25 16:17:14.933 GMT [195] LOG:  database system was shut down in recovery at 2020-09-25 16:17:13 GMT
pg-1_1    | 2020-09-25 16:17:14.934 GMT [195] LOG:  entering standby mode
pg-1_1    | 2020-09-25 16:17:14.938 GMT [195] LOG:  redo starts at 0/4000028
pg-1_1    | 2020-09-25 16:17:14.938 GMT [195] LOG:  consistent recovery state reached at 0/50005E0
pg-1_1    | 2020-09-25 16:17:14.938 GMT [195] LOG:  invalid record length at 0/50005E0: wanted 24, got 0
pg-1_1    | 2020-09-25 16:17:14.940 GMT [193] LOG:  database system is ready to accept read only connections
pg-1_1    | 2020-09-25 16:17:14.948 GMT [199] LOG:  started streaming WAL from primary at 0/5000000 on timeline 1
pg-0_1    | 2020-09-25 16:17:14.948 GMT [294] LOG:  standby "walreceiver" is now a synchronous standby with priority 1
pg-1_1    |  done
pg-1_1    | server started
pg-1_1    | postgresql-repmgr 16:17:15.00 INFO  ==> ** Starting repmgrd **
pg-1_1    | [2020-09-25 16:17:15] [NOTICE] repmgrd (repmgrd 5.1.0) starting up
pg-1_1    | INFO:  set_repmgrd_pid(): provided pidfile is /opt/bitnami/repmgr/tmp/repmgr.pid
pg-1_1    | [2020-09-25 16:17:15] [NOTICE] starting monitoring of node "pg-1" (ID: 1001)
pgpool_1  | 2020-09-25 16:17:17: pid 145: LOG:  health check retrying on DB node: 1 succeeded
pg-0_1    | [2020-09-25 16:17:19] [NOTICE] new standby "pg-1" (ID: 1001) has connected
pgpool_1  | 2020-09-25 16:17:37: pid 143: ERROR:  Failed to check replication time lag
pgpool_1  | 2020-09-25 16:17:37: pid 143: DETAIL:  No persistent db connection for the node 1
pgpool_1  | 2020-09-25 16:17:37: pid 143: HINT:  check sr_check_user and sr_check_password
pgpool_1  | 2020-09-25 16:17:37: pid 143: CONTEXT:  while checking replication time lag
pgpool_1  | 2020-09-25 16:22:52: pid 1: LOG:  child process with pid: 162 exits with status 256
pgpool_1  | 2020-09-25 16:22:52: pid 1: LOG:  fork a new child process with pid: 723
pgpool_1  | 2020-09-25 16:25:44: pid 1: LOG:  child process with pid: 132 exits with status 256
pgpool_1  | 2020-09-25 16:25:44: pid 1: LOG:  fork a new child process with pid: 1010
pgpool_1  | 2020-09-25 16:26:05: pid 1: LOG:  child process with pid: 139 exits with status 256
pgpool_1  | 2020-09-25 16:26:05: pid 1: LOG:  fork a new child process with pid: 1045
pgpool_1  | 2020-09-25 16:26:55: pid 1: LOG:  child process with pid: 138 exits with status 256
pgpool_1  | 2020-09-25 16:26:55: pid 1: LOG:  fork a new child process with pid: 1133
pgpool_1  | 2020-09-25 16:27:46: pid 1: LOG:  child process with pid: 164 exits with status 256
pgpool_1  | 2020-09-25 16:27:46: pid 1: LOG:  fork a new child process with pid: 1219
pgpool_1  | 2020-09-25 16:27:56: pid 1: LOG:  child process with pid: 133 exits with status 256
pgpool_1  | 2020-09-25 16:27:56: pid 1: LOG:  fork a new child process with pid: 1237
pgpool_1  | 2020-09-25 16:27:56: pid 1: LOG:  child process with pid: 723 exits with status 256
pgpool_1  | 2020-09-25 16:27:56: pid 1: LOG:  fork a new child process with pid: 1238
pgpool_1  | 2020-09-25 16:28:47: pid 1: LOG:  child process with pid: 136 exits with status 256
pgpool_1  | 2020-09-25 16:28:47: pid 1: LOG:  fork a new child process with pid: 1323
pgpool_1  | 2020-09-25 16:30:39: pid 1: LOG:  child process with pid: 134 exits with status 256
pgpool_1  | 2020-09-25 16:30:39: pid 1: LOG:  fork a new child process with pid: 1511
pgpool_1  | 2020-09-25 16:30:39: pid 1: LOG:  child process with pid: 130 exits with status 256
pgpool_1  | 2020-09-25 16:30:39: pid 1: LOG:  fork a new child process with pid: 1512
pgpool_1  | 2020-09-25 16:30:59: pid 1: LOG:  child process with pid: 1010 exits with status 256

```

**Describe the results you expected:**

Databases start, are connected to pgpool, with empty database schemas as defined in https://github.com/ksonda/pidpool/blob/master/scripts/postgresql.sql

**Additional information you deem important (e.g. issue happens only occasionally):**

**Version**

- Output of `docker version`:

```
Client:
 Version:           19.03.6
 API version:       1.40
 Go version:        go1.12.17
 Git commit:        369ce74a3c
 Built:             Fri Feb 28 23:45:43 2020
 OS/Arch:           linux/amd64
 Experimental:      false
Server:
 Engine:
  Version:          19.03.6
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.17
  Git commit:       369ce74a3c
  Built:            Wed Feb 19 01:06:16 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.3.3-0ubuntu1~18.04.2
  GitCommit:        
 runc:
  Version:          spec: 1.0.1-dev
  GitCommit:        
 docker-init:
  Version:          0.18.0
  GitCommit:   
```

- Output of `docker info`:

```
Client:
 Debug Mode: false
Server:
 Containers: 3
  Running: 3
  Paused: 0
  Stopped: 0
 Images: 2
 Server Version: 19.03.6
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 
 runc version: 
 init version: 
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.4.0-1025-gcp
 Operating System: Ubuntu 18.04.5 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 7.775GiB
 Name: pidpool
 ID: NQP4:QCEV:3WOH:T3VG:CTFX:AQUL:HBZJ:I6ZR:LCOV:PPOL:EEFA:ZMFZ
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.17.1, build unknown
docker-py version: 2.5.1
CPython version: 2.7.17
OpenSSL version: OpenSSL 1.1.1  11 Sep 2018
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**

Ubuntu 18.04 LTS host machine on GCP with 8GB ram.

**Comment by marcosbc** [comment]
Hi @ksonda, it looks like the `postgresql-repmgr` is not fully supporting init scripts, because it shuts down right after the entries are inserted in the DB without waiting for it to replicate:

```
pg-0_1    | 2020-09-29 17:16:17.926 GMT [221] LOG:  received smart shutdown request
pg-0_1    | 2020-09-29 17:16:17.930 GMT [221] LOG:  background worker "logical replication launcher" (PID 230) exited with exit code 1
pg-0_1    | 2020-09-29 17:16:19.856 GMT [290] FATAL:  the database system is shutting down
```

Refer to [this](https://github.com/bitnami/bitnami-docker-postgresql-repmgr/blob/master/12/debian-10/rootfs/opt/bitnami/scripts/libpostgresql.sh#L650) file for more info in the logic.

It seems like we can workaround it with this:

```diff
diff --git a/docker-compose.yml b/docker-compose.yml
index d1626f9..70668da 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -2,6 +2,7 @@ version: '2.1'
 services:
   pg-0:
     image: docker.io/bitnami/postgresql-repmgr:11-debian-10
+    restart: always
     ports:
       - 5432
     volumes:
@@ -20,6 +21,7 @@ services:
       - REPMGR_NODE_NETWORK_NAME=pg-0
   pg-1:
     image: docker.io/bitnami/postgresql-repmgr:11-debian-10
+    restart: always
     ports:
       - 5432
     volumes:
```

To solve this, I think we may need to revisit the logic of the current container image like this, as it looks like a forceful PostgreSQL shutdown is causing these issues to happen:

```diff
diff --git a/images/generic/lib/libpostgresql.sh b/images/generic/lib/libpostgresql.sh
index 8a84090fc488b..d8438020da4c2 100644
--- a/images/generic/lib/libpostgresql.sh
+++ b/images/generic/lib/libpostgresql.sh
@@ -689,7 +689,7 @@ postgresql_custom_init_scripts() {
 #########################
 postgresql_stop() {
     info "Stopping PostgreSQL..."
-    stop_service_using_pid "$POSTGRESQL_PID_FILE"
+    PGDATA="$POSTGRESQL_DATA_DIR" pg_ctl stop -w
 }

 ########################
```

Could you try to build a Bitnami PostgreSQL Docker image with that change to see if it works on your side?

**Comment by ksonda** [comment]
Thank you, that appears to do the trick!

**Comment by ksonda** [comment]
Random question though. To scale up the standby nodes when using docker-compose, do you use the --scale option or do you recommend adding configurations for pg-2, pg-3, etc to the docker-compose.yml ?

**Comment by marcosbc** [comment]
@ksonda AFAIK we don't support the `--scale` option because it would require to implement some kind of service discovery logic in the container entrypoints. For now, it's probably best if you add those manually.

However, if you were interested in getting it to work and would be willing to look into implementing it for this specific container, we'd be glad to accept such contribution.

In the end did you solve it via the change in the .sh script, or by adding "restart:" to the docker-compose?

**Comment by ksonda** [comment]
@marcosbc Regarding `--scale` Thanks, good to know. I may look into it in the next couple of months but can't invest a lot of time in the very short term.  

Regarding this issue, I actually had to change both. Adding `restart:` alone, the intialization scripts still never ran, its just the postgres instances would restart on exit after a connection fail. Adding the `.sh` alone worked, but then the db containers failed to restart when I stressed them with requests designed to overwhelm them.
